### PR TITLE
Us 66 add throw dice action dummy

### PR DESCRIPTION
**Pull request by andrii-r**

### CODE REVIEW

File: src/main/java/catan/controllers/PlayController.java [hunk 1]
```diff
@@ -29,7 +32,10 @@ public void buildRoad(@RequestParam(value = "token", required = false) String to
                           @RequestParam("edgeId") String edgeId) throws AuthenticationException, GameException, PlayException {
         UserBean user = authenticationService.authenticateUserByToken(token);
 
-        playService.buildRoad(user, gameId, edgeId);
+        HashMap<String, String> params = new HashMap<String, String>();
```
Comment (ypetrenko) [inline src/main/java/catan/controllers/PlayController.java:35]: TBD: do we need a HashMap variable here or we can push into processAction() String param instead?
For now we push only one parameter into controllers and it has a String type

File: src/main/java/catan/services/impl/PlayServiceImpl.java [hunk 2]
```diff
@@ -41,86 +44,82 @@
...
+            case THROW_DICE:
+                throwDice(game);
+                break;
+        }
+    }
 
-        preparationStageUtil.updateCurrentCycleBuildingNumber(game);
+    private void buildRoad(UserBean user, GameBean game, String edgeId) throws PlayException, GameException {
+        EdgeBean edgeToBuildOn = (EdgeBean) buildUtil.getValidMapElementByIdToBuildOn(edgeId, new ArrayList<MapElement>(game.getEdges()));
+        buildUtil.validateUserCanBuildRoanOnEdge(user, edgeToBuildOn);
```
Comment (ypetrenko) [inline src/main/java/catan/services/impl/PlayServiceImpl.java]: bad naming. "Road", not "Roan"

File: src/main/java/catan/services/util/play/PlayUtil.java [hunk 3]
```diff
@@ -48,7 +47,7 @@ public void updateAvailableUserActions(GameBean game) throws GameException {
                 preparationStageUtil.updateAvailableUserActions(game);
```
Comment (ypetrenko) [inline src/main/java/catan/services/util/play/PlayUtil.java]: this method also update actions for all users, so name should be the same as for main stage: updateAvailableActionsForAllUsers

File: src/main/java/catan/services/util/play/PreparationStageUtil.java [hunk 4]
```diff
@@ -124,16 +120,16 @@ private boolean isEndOfCycle(GameBean game) {
         return firstPlayer && !oddCycle || lastPlayer && oddCycle;
     }
 
-    private GameUserActionCode getCurrentInitialAction(GameBean game) {
+    private GameUserActionCode getCurrentInitialBuildAction(GameBean game) {
```
Comment (ypetrenko) [inline src/main/java/catan/services/util/play/PreparationStageUtil.java]: method returns END_TURN in some cases, so naming "InitialBuildAction" is incorrect

File: src/test/java/catan/services/impl/PlayServiceImplTest.java [hunk 5]
```diff
@@ -756,22 +885,220 @@ public void shouldUpdateVictoryPointsOnMultipleBuildingsInMainStage() throws Exc
...
+        params = new HashMap<String, String>();
+        params.put("nodeId", "3");
+
+        playService.processAction(GameUserActionCode.BUILD_CITY, gameUser1.getUser(), "1", params);
 
         assertEquals(2, gameUser1.getAchievements().getDisplayVictoryPoints());
     }
 
+    @Test
+    public void shouldPassWhenBuildCityInMainStageOnlyAfterDiceThrown() throws PlayException, GameException{
```
Comment (ypetrenko) [inline src/test/java/catan/services/impl/PlayServiceImplTest.java]: this is two different cases:
shouldFailWhenBuildCityInMainStageBeforeDiceThrown
shouldPassWhenBuildCityInMainStageAfterDiceThrown

File: src/test/java/catan/services/impl/PlayServiceImplTest.java [hunk 6]
```diff
@@ -206,7 +217,11 @@ public void shouldFailWhenBuildRoadNotNearOwnNeighbourCityOrRoad() throws GameEx
...
             // WHEN
-            playService.buildRoad(gameUser1.getUser(), "1", "7");
+
+            HashMap<String, String> params = new HashMap<String, String>();
+            params.put("edgeId", "7");
+
+            playService.processAction(GameUserActionCode.BUILD_ROAD, gameUser1.getUser(), "1", params);
             fail("PlayException with error code '" + PlayServiceImpl.ERROR_CODE_ERROR + "' should be thrown");
         } catch (PlayException e) {
             // THEN
```
Comment (ypetrenko) [inline src/test/java/catan/services/impl/PlayServiceImplTest.java:227]: GOTO LINE 244: forgot to fix it.

File: src/main/java/catan/controllers/PlayController.java (hunk 1 above)
Comment (andrii-r) [inline src/main/java/catan/controllers/PlayController.java:35]: decided to leave as is until new solution is invented

File: src/main/java/catan/services/impl/PlayServiceImpl.java (hunk 2 above)
Comment (andrii-r) [inline src/main/java/catan/services/impl/PlayServiceImpl.java]: done

File: src/main/java/catan/services/util/play/PlayUtil.java (hunk 3 above)
Comment (andrii-r) [inline src/main/java/catan/services/util/play/PlayUtil.java]: done

File: src/main/java/catan/services/util/play/PreparationStageUtil.java (hunk 4 above)
Comment (andrii-r) [inline src/main/java/catan/services/util/play/PreparationStageUtil.java]: anyway, it is not InitialAction)

File: src/test/java/catan/services/impl/PlayServiceImplTest.java (hunk 6 above)
Comment (andrii-r) [inline src/test/java/catan/services/impl/PlayServiceImplTest.java:227]: wat?

Comment (ypetrenko) [inline src/test/java/catan/services/impl/PlayServiceImplTest.java:227]: that was a result of merging. merge was first before code review, so i decided you just forgot to fix String param to HashMap. already fixed

File: src/main/java/catan/services/util/play/PreparationStageUtil.java (hunk 4 above)
Comment (andrii-r) [inline src/main/java/catan/services/util/play/PreparationStageUtil.java]: renamed to getCurrentActionCode

File: src/test/java/catan/services/impl/PlayServiceImplTest.java (hunk 5 above)
Comment (andrii-r) [inline src/test/java/catan/services/impl/PlayServiceImplTest.java]: the idea was to show in one test case that user can build city ONLY after dice is thrown, but ok, I have splited it to 2 differnt test cases

**Pull request merged**